EventsExpress/ClientApp/src/containers/edit-event.js

```diff
@@ -1,21 +1,28 @@
 ﻿import React, { Component } from 'react';
 import EventForm from '../components/event/event-form';
-import { withRouter } from "react-router";
+import { withRouter } from 'react-router';
 import { connect } from 'react-redux';
-import { getFormValues } from 'redux-form';
+import { getFormValues, SubmissionError } from 'redux-form';
 import { edit_event } from '../actions/event/event-add-action';
 import { setSuccessAllert } from '../actions/alert-action';
-import { validate } from './event-edit-validate-form '
-import { validateEventForm } from './event-validate-form'
-import Button from "@material-ui/core/Button";
+import { validate } from './event-edit-validate-form ';
+import { validateEventForm } from './event-validate-form';
+import Button from '@material-ui/core/Button';
+import { buildValidationState } from '../components/helpers/action-helpers';
 
 class EditEventWrapper extends Component {
 
     onSubmit = async (values) => {
-        await this.props.edit_event({ ...validateEventForm(values), user_id: this.props.user_id, id: this.props.event.id });
-        this.props.alert('Your event has been successfully saved!');    
+        await this.props.edit_event({
+            ...validateEventForm(values),
+            user_id: this.props.user_id,
+            id: this.props.event.id
+        }, async response => {
+            throw new SubmissionError(await buildValidationState(response));
+        });
+        this.props.alert('Your event has been successfully saved!');
```

Comment:
Shouldn't this alert be in the onSuccess?

Reply:
Yes